utils/utils.go

```diff
@@ -1111,19 +1150,37 @@ func getOrInitProbe(probe *corev1.Probe) *corev1.Probe {
 func patchFileBasedProbe(defaultProbe *corev1.Probe, instanceProbe *corev1.Probe, scriptName string, probeFile string) *corev1.Probe {
 	defaultProbe = getOrInitProbe(defaultProbe)
 	instanceProbe = getOrInitProbe(instanceProbe)
-	isExecConfigured := instanceProbe.Exec != nil
+	isExecConfigured := instanceProbe.Exec != nil // this flag allows the user to override the ExecAction object to bring their own custom file-based health check
 	instanceProbe = rcoutils.CustomizeProbeDefaults(instanceProbe, defaultProbe)
 	if !isExecConfigured {
 		configureFileBasedProbeExec(instanceProbe, scriptName, probeFile)
 	}
 	return instanceProbe
 }
 
+// Modifies the PodTemplateSpec to use Liberty file-based health checks in the "app" container
 func CustomizePodSpecFileBasedProbes(pts *corev1.PodTemplateSpec, instance *wlv1.WebSphereLibertyApplication) {
 	if !IsFileBasedProbesEnabled(instance) {
 		return
 	}
 	appContainer := rcoutils.GetAppContainer(pts.Spec.Containers)
+	customizeFileBasedProbes(appContainer, instance)
+}
+
+// Modifies the Knative Service instance to use Liberty file-based health checks in the "user-container" container
+func CustomizeKnativeServiceFileBasedProbes(ksvc *servingv1.Service, instance *wlv1.WebSphereLibertyApplication) {
+	if !IsFileBasedProbesEnabled(instance) {
+		return
+	}
+	if len(ksvc.Spec.Template.Spec.Containers) == 0 {
+		return
+	}
+	appContainer := &ksvc.Spec.Template.Spec.Containers[0]
+	customizeFileBasedProbes(appContainer, instance)
+}
+
+// A helper function to customize Liberty file-based health checks on top of the MicroProfile probe defaults in GetDefaultStartupProbe
+func customizeFileBasedProbes(appContainer *corev1.Container, instance *wlv1.WebSphereLibertyApplication) {
 	if appContainer == nil {
 		return
 	}
```

Comment:
The if-checks in OLO [here](https://github.com/OpenLiberty/open-liberty-operator/pull/800/files#diff-0df82f8689d3e8d881be0653ab4e9f8221c5835ec95a14930d11669e9c7f07efR1113) need to be added to WLO.

Reply:
Thanks for catching that, added a new commit for setting probes invidually.